tests/test_fairworkflow.py

```diff
@@ -347,6 +347,7 @@ def my_workflow(in1, in2, in3):
         fw = FairWorkflow.from_function(my_workflow)
 
         assert isinstance(fw, FairWorkflow)
+        assert len(fw._steps) == 4
```

Comment:
This is nice, although I suppose this wouldn't work if we 'reused' a step? I think that's already its own issue though.

Reply:
If this behavior changes we shall update the test.